exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpSender.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.http;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.function.Consumer;
+
+/**
+ * An abstraction for sending HTTP requests and handling responses.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ *
+ * @see HttpExporter
+ * @see HttpExporterBuilder
+ */
+public interface HttpSender {
```

Comment:
I've had to create this thing in multiple projects now. It's almost like the java ecosystem needs an http client equivalent of slf4j. 🤔

Reply:
And be based on an interface, not an abstract class.  😿